### ISSUE TITLE
ETEF instances for flows between in and out feature groups in leaf components

### ISSUE THREAD

**Issue by reteprelief** [osate/osate2-core#794]
If a leaf component has incoming and outgoing feature group then we currently generate all possible flows from each feature of the incoming FD to each feature in the outgoing FG.
Users can specify multiple flow implementations for a flow path between FGs that identify a flow for each element in the feature group. The flow implementation may just refer to the incoming FG.feature and outgoing FG.feature. as a flow path.
We should interpret those flow implementations to identify the relevant outgoing feature for which we are finding outgoing connections.
```
package FeatureGroupflows
public
	feature group sensoroutput
		features
			speedreading: out data port;
			altitudereading: out data port;
	end sensoroutput;
	feature group commandinput
		features
			speedcmd: in data port;
			altitudecmd: in data port;
	end commandinput;
	abstract sensors
		features
			readings: feature group sensoroutput;
		flows 
			sensesrc: flow source readings;
	end sensors;
	abstract actuators
		features
			commands: feature group commandinput;
		flows 
			actuatedst: flow sink commands;
	end actuators;
	abstract controllers
		features
			readings: feature group inverse of sensoroutput;
			commands: feature group inverse of commandinput;
		flows
			through: flow path readings -> commands;
			throughs: flow path readings.speedreading -> commands.speedcmd;
			througha: flow path readings.altitudereading -> commands.altitudecmd;
	end controllers;
	
	system topsystem
	
	end topsystem;
	
	system implementation topsystem.tier1
		subcomponents
			sense: abstract sensors;
			actuate: abstract actuators;
			control: abstract controllers;
		connections
			stoc: feature group sense.readings -> control.readings;
			ctoa: feature group control.commands -> actuate.commands;
		flows
			etef1: end to end flow sense.sensesrc -> stoc -> control.through -> ctoa -> actuate.actuatedst;
	end topsystem.tier1;
	
	device speedsensor
		features
			speedreading: out data port;
		flows
			sensesrc: flow source speedreading;
	end speedsensor;
	device altitudesensor
		features
			altitudereading: out data port;
		flows
			sensesrc: flow source altitudereading;
	end altitudesensor;
	
	device speedactuator
		features
			speedcmd: in data port;
		flows
			actuatedst: flow sink speedcmd;
	end speedactuator;
	device altitudeactuator
		features
			altitudecmd: in data port;
		flows
			actuatedst: flow sink altitudecmd;
	end altitudeactuator;

	
	device speedcontroller
		features
			speedreading: in data port;
			speedcmd: out data port;
		flows
			speedcontrol: flow path speedreading -> speedcmd;
	end speedcontroller;
	device altitudecontroller
		features
			altitudereading: in data port;
			altitudecmd: out data port;
		flows
			altcontrol: flow path altitudereading -> altitudecmd;
	end altitudecontroller;
	
	abstract implementation sensors.impl
		subcomponents
			speed: device speedsensor;
			altitude: device altitudesensor;
		connections
			cs1: port speed.speedreading -> readings.speedreading;
			cs2: port altitude.altitudereading -> readings.altitudereading;
		flows
			sensesrc: flow source speed.sensesrc -> cs1 -> readings;
			sensesrc: flow source altitude.sensesrc -> cs2 -> readings;
	end sensors.impl;
	
	abstract implementation actuators.impl
		subcomponents
			speed: device speedactuator;
			altitude: device altitudeactuator;
		connections
			cs1: port commands.speedcmd -> speed.speedcmd;
			cs2: port commands.altitudecmd -> altitude.altitudecmd;
		flows
			actuatedst: flow sink commands -> cs1 -> speed.actuatedst;
			actuatedst: flow sink commands -> cs2 -> altitude.actuatedst;
	end actuators.impl;
	
	abstract implementation controllers.impl
		flows
			-- these flow path implementations map each incoming feature to one outgoing feature.
			-- This outgoing feature determines how the ETEF continues (instead of one-to-all).
			through: flow path readings.speedreading -> commands.speedcmd;
			through: flow path readings.altitudereading -> commands.altitudecmd;
	end controllers.impl;
	
	system implementation topsystem.tier2 extends topsystem.tier1
		subcomponents
			sense: refined to abstract sensors.impl;
			actuate: refined to abstract actuators.impl;
			control: refined to abstract controllers.impl;
	end topsystem.tier2;
	
end FeatureGroupflows;
```

**Comment by lwrage** [comment]
This requires a metamodel change and the standard needs to be updated (saeaadl/aadlv2.2#4)

**Comment by lwrage** [comment]
@reteprelief I assume that we will allow only one nesting level - feature in feature group `fg.f` - but no deeper nesting such as `fg.f1.f2` in a flow implementation.

**Comment by lwrage** [comment]
This is a straight forward change in the meta-model and parser
- [ ] Add endpoints to flow implementation
- [ ] Repair validation that checks if connection and flow are connected
End to end flow instances reference the elements that make up the flows. If we do not instantiate flow implementations we can continue to use use the corresponding flow specification instance. (All flow properties are associated with flow specifications!)
The flow implementations will be used only to find the pairs of features to be connected in the end to end flow. Without a flow implementation we generate all possible pairs; with flow implementations we create a subset.
- [ ] Adapt instantiation of end to end flows

**Comment by lwrage** [comment]
Fixed by merging #813
